source/core/src/main/com/csse3200/game/entities/Entity.java

```diff
@@ -467,6 +468,9 @@ public void read(Json json, JsonValue jsonMap) {
                             .get(ShipLightComponent.class.getSimpleName());
                     shipLightComponent.read(json, shipLightComponentJson);
 
+                    ShipProgressComponent progressComponent = ship.getComponent(ShipProgressComponent.class);
+                    progressComponent.read(json, jsonMap.get("components").get(ShipProgressComponent.class.getName()));
```

Comment:
this should be the same as what you use in `write`, i.e. `ShipProgressComponent.class.getSimpleName()`